dali/pipeline/operators/reader/reader_op.h

```diff
@@ -47,217 +47,191 @@ namespace dali {
 template <typename Backend, typename LoadTarget, typename ParseTarget = LoadTarget>
 class DataReader : public Operator<Backend> {
  public:
-  inline explicit DataReader(const OpSpec& spec) :
-    Operator<Backend>(spec),
-  prefetched_batch_ready_(false),
-  prefetch_ready_workers_(false),
-  finished_(false),
-  samples_processed_(0),
-  batch_stop_(false),
-  prefetch_error_(nullptr) {
-  }
+  inline explicit DataReader(const OpSpec& spec)
+      : Operator<Backend>(spec),
+        finished_(false),
+        prefetch_queue_depth_(spec.GetArgument<int>("prefeth_queue_depth")),
+        prefetched_batch_queue_(prefetch_queue_depth_),
+        curr_batch_consumer_(0),
+        curr_batch_producer_(0),
+        consumer_cycle_(false),
+        producer_cycle_(false),
+        samples_processed_(0) {}
 
   ~DataReader() noexcept override {
     StopPrefetchThread();
-    for (size_t i = 0; i < prefetched_batch_.size(); ++i) {
-      // return unconsumed batches
-      if (prefetched_batch_[i]) {
-        loader_->ReturnTensor(prefetched_batch_[i]);
+    for (auto &batch : prefetched_batch_queue_) {
+      for (auto &sample : batch) {
+        if (sample) loader_->ReturnTensor(sample);
```

Comment:
Personally I don't like wrapping that kind of code to one line, but that's only my opinion.

Reply:
changed